src/client/game.cpp

```diff
@@ -3000,6 +3015,20 @@ void Game::handleClientEvent_SetSky(ClientEvent *event, CameraOrientation *cam)
 	// Orbit Tilt:
 	sky->setBodyOrbitTilt(event->set_sky->body_orbit_tilt);
 
+	// fog
+	// do not override a potentially smaller client setting.
+	sky->setFogDistance(event->set_sky->fog_distance);
+
+	// if the fog distance is reset, switch back to the client's viewing_range
+	if (event->set_sky->fog_distance < 0)
+		draw_control->wanted_range = g_settings->getS16("viewing_range");
+
+	if (event->set_sky->fog_start >= 0)
+		sky->setFogStart(rangelim(event->set_sky->fog_start, 0.0f, 0.99f));
```

Comment:
`rangelim` keeps NaNs.
This only doesn't matter here because of the if condition. A comment would be good, e.g.:
```suggestion
	if (event->set_sky->fog_start >= 0) // also not NaN
		sky->setFogStart(rangelim(event->set_sky->fog_start, 0.0f, 0.99f));
```

Reply:
Aren't there many other places where we could possibly have NaN? We don't put a comment on each of them.
Happy to do so, just seems a bit over the top.

Reply:
I would find it good if we did nan handling consistently and explicitly.
But it's fine if you don't deem it necessary here, I guess.